glean-core/ios/Glean/Glean.swift

```diff
@@ -263,6 +263,77 @@ public class Glean {
+    ///
+    /// - `logPings`: If "true", will cause pings that are submitted to also be echoed to the device's log
+    /// - `tagPings`:  This command expects a string to tag the pings with and redirects them to the Glean Debug View
+    /// - `sendPing`: This command expects a string name of a ping to force immediate collection and submission of.
+    ///
+    /// The structure of the custom URL uses the following format:
+    ///
+    /// `<protocol>://glean?<command 1>=<paramter 1>&<command 2>=<parameter 2> ...`
+    ///
+    /// Where:
+    ///
+    /// - `<protocol>://` is the "Url Scheme" that has been added for the app and doesn't matter to Glean.
+    /// - `glean` is required for the Glean SDK to recognize the command is meant for it to process.
+    /// - `?` indicating the beginning of the query.
+    /// - `<command>=<parameter>` are instances of the commands listed above  separated by `&`.
+    ///
+    /// There are a few things to consider when creating the custom URL:
+    ///
+    /// - Invalid commands will trigger an error and be ignored.
+    /// - Not all commands are requred to be encoded in the URL, you can mix and match the commands that you need.
+    /// - Special characters should be properly URL encoded and escaped since this needs to represent a valid URL.
+    public func handleCustomUrl(url: URL) {
+        guard url.host?.removingPercentEncoding == "glean",
+            let components = URLComponents(url: url, resolvingAgainstBaseURL: true),
+            let params = components.queryItems else {
+            logger.error("Error parsing query parameters")
+            return
+        }
+
+        for param in params {
+            switch param.name {
+            case "tagPings":
+                if let pingTag = param.value,
+                    let pingTagPattern = try? NSRegularExpression(pattern: "^[a-zA-Z0-9-]{1,20}$") {
+                    _ = pingTagPattern.matches(
+                        in: pingTag,
+                        options: [],
+                        range: NSRange(location: 0, length: pingTag.count)
+                    ).map { _ in
+                        configuration?.pingTag = pingTag
+                        logger.debug("Pings tagged with: \(pingTag)")
+                    }
+                } else {
+                    configuration?.pingTag = nil
+                    logger.debug("No value passed to tagPings, clearing current value")
+                }
+            case "logPings":
+                configuration?.logPings = param.value == "true" ? true : false
+                logger.debug("Log pings set to: \(configuration!.logPings)")
+            case "sendPing":
+                if let pingName = param.value {
+                    // Check to see if the passed ping name matches a registered ping name
+                    // before sending the ping
+                    if pingTypeQueue.contains(where: { $0.name == pingName }) {
```

Comment:
Do we need this? Does Android do this?
Glean-core should log an error if you're trying to send an unknown ping (which might be very helpful for debugging)

Reply:
Oh! We should also defer the sending until after we looped through all parameters.
Otherwise we might first send a ping and only then parse the tag and later set it.

Reply:
> Do we need this?

I don't know, [you are the one that asked for it 😉 ](https://github.com/mozilla/glean/pull/409#discussion_r349500378).  This was just the simplest way I found to accomplish it.

> Does Android do this?

I also [filed a bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1599483) to replicate this behavior in Android.

Reply:
Now I asked this question in both directions! (: 
Guess we keep it, it's debugging so I don't even care much about performance.